### PR TITLE
v.0.0.3 periodic data items

### PR DESCRIPTION
**Pull request by jeffhuth-bytecode**
# Description of change
Re-work periodic_data_standardized and unique key + bookmarking strategy. Add periodic_data_calculated - new endpoint.

# Manual QA steps
Ran singer-discover, singer-check-tap, and target-stitch for all endpoints. Reviewed data in downstream target DWH.
 
# Risks
Low - new tap, new endpoint. Currently in Alpha testing w/ Bytecode (110918) and Felicis (156601).
 
# Rollback steps
Revert to v.0.0.2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,4 +153,4 @@ This tap:
     ```
 ---
 
-Copyright &copy; 2020 Stitch
+Copyright &copy; 2019 Stitch
```
Comment (KAllan357) [inline README.md:156]: This should stay 2020

File: tap_ilevel/singer_operations.py [hunk 2]
```diff
@@ -83,7 +87,7 @@ def write_bookmark(state, stream, value):
     if 'bookmarks' not in state:
         state['bookmarks'] = {}
     state['bookmarks'][stream] = value
-    LOGGER.info('Write state for stream: %s, value: %s', stream, value)
+    LOGGER.info('Write state for stream: {}, value: {}'.format(stream, value))
```
Comment (KAllan357) [inline tap_ilevel/singer_operations.py:90]: Using `format` violates pylint for these logger changes. This should stay as it was.

File: tap_ilevel/transform.py [hunk 3]
```diff
@@ -1,42 +1,19 @@
...
-        'records': records
-    }
-    transformed_json = convert_json(records)
-    return transformed_json['records']
+    transformed_json = humps.decamelize(records)
+    return transformed_json
+
+
+# Create MD5 hash key for data element
+def hash_data(data):
```
Comment (KAllan357) [inline tap_ilevel/transform.py:15]: This function does not seem to be referenced anywhere?

File: tap_ilevel/transform.py [hunk 4]
```diff
@@ -1,42 +1,19 @@
...
-    new_arr = []
-    for i in arr:
-        if isinstance(i, list):
-            new_arr.append(convert_array(i))
-        elif isinstance(i, dict):
-            new_arr.append(convert_json(i))
-        else:
-            new_arr.append(i)
-    return new_arr
+#!/usr/bin/env python3
```
Comment (KAllan357) [inline tap_ilevel/transform.py:1]: This seems strange - is it necessary?

File: tap_ilevel/transform.py (hunk 3 above)
Comment (jeffhuth-bytecode) [reply tap_ilevel/transform.py:15]: `hash_data` is imported and used in `sync.py` and `ilevel_api.py` to simplify a multi-dimension key.

File: tap_ilevel/transform.py (hunk 4 above)
Comment (jeffhuth-bytecode) [reply tap_ilevel/transform.py:1]: Ok - removed. When is this header needed?

File: README.md (hunk 1 above)
Comment (jeffhuth-bytecode) [reply README.md:156]: Ok - fixed.

File: tap_ilevel/singer_operations.py (hunk 2 above)
Comment (jeffhuth-bytecode) [reply tap_ilevel/singer_operations.py:90]: Ok - changed back.

**Pull request merged**